### DISCUSSION TITLE
How can I specify the language by file?

### DISCUSSION THREAD

**Discussion by jankap** [valentjn/vscode-ltex#465] [Q&A] [answered]
Awesome tool, thank you!

However, I switch between German (de-DE) and English (~~en-EN~~ en-US) a lot of times per day. Is there an option to specify the language by file or any other workaround. As far as I understand, `auto` mode is not a good idea?

Thanks!

**Comment by valentjn** [comment, marked as answer]
[Please read the FAQ.](https://valentjn.github.io/ltex/faq.html#how-can-i-check-multiple-languages-at-once)

Also, `en-EN` is not a [valid language](https://valentjn.github.io/ltex/settings.html#ltexlanguage).

**Reply by jankap** [reply]
True, meant `en-US`. Thanks for the link, don't know why I didn't find that, I did see the offline install docs though. Strange. Is the README in VSCode the same as here?

Anyway, using the magic commands (since I'm using markdown currently) is fine, thanks!

Edit, I already have a language defined in the yaml header of my markdown file. Is there any plan to support parsing that? :)
![image](https://user-images.githubusercontent.com/12893844/142652969-cc727ceb-49a7-40f8-8d57-154e657d7d52.png)

**Reply by valentjn** [reply]
@jankap The README in the repo is the same as in VS Code.

Is there a specification that contains the name `lang` of the key and the possible values, together with the meaning of the values?

Because it might be necessary to map values to LanguageTool languages. For example, `de` wouldn't be a wise choice to use for LanguageTool as there's no spell-checking with this (only grammar checking), in contrast to `de-DE`.

**Reply by jankap** [reply]
Not exactly a specification, but at least some information can be found here: https://pandoc.org/MANUAL.html#language-variables
And indeed, `de-DE` can be set, too and it gets forwarded to babel when compiled to Latex. However, I'm using the HTML renderer currently.
The `---` fences are called YAML header in Markdown/Pandoc. Maybe you can teach ltex to parse that config location, too. 

However, I'm fine with the current solution, so thanks a lot for your help!

**Reply by valentjn** [reply]
@jankap Thanks for the link. This is implemented now and will be included in the next release.

Edit: Feature released in 13.1.0.

**Reply by jankap** [reply]
Awesome, thank you!
